tests/workflow_task_reported_problems_test.go

```diff
@@ -45,6 +43,22 @@ func (s *WFTFailureReportedProblemsTestSuite) simpleActivity() (string, error) {
 	return "done!", nil
 }
 
+// workflowWithSignalsThatFails creates a workflow that listens for signals and fails on each workflow task.
+// This is used to test that the TemporalReportedProblems search attribute is not incorrectly removed
+// when signals keep coming in despite continuous workflow task failures.
+func (s *WFTFailureReportedProblemsTestSuite) workflowWithSignalsThatFails(ctx workflow.Context) (string, error) {
+	// If we should fail, signal ourselves (creating a side effect) and immediately panic.
+	// This will create buffered.
+	if s.shouldFail.Load() {
+		// Signal ourselves to create buffered events
+		_ = workflow.SignalExternalWorkflow(ctx, workflow.GetInfo(ctx).WorkflowExecution.ID, "", "test-signal", "self-signal")
```

Comment:
This doesn't create buffered events, you need to do it from a `workflow.SideEffect`: https://pkg.go.dev/go.temporal.io/sdk/workflow#hdr-SideEffect_API

Reply:
You will need to use a temporal client to issue the signal, and make sure you don't ignore errors, we should be able to tell that the signal was successfully submitted.